### ISSUE TITLE
Allow for WP Constants when checking prefixes

### ISSUE THREAD

**Issue by GaryJones** [WordPress-Coding-Standards/WordPress-Coding-Standards#1311]
```
define( 'SCRIPT_DEBUG', true );`
```
will throw an error of `WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedConstantFound`.

The [checking code](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/c0e29132df7c8215423ac998ee97c5e9aef7d469/WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php#L285-L305) doesn't have any allowance for constants that WordPress uses that might be defined in plugins and themes.

There is some code that [ignores supergloabls and WP_Globals](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/c0e29132df7c8215423ac998ee97c5e9aef7d469/WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php#L709-L712), but not WP Constants.

Reference of [default constants](https://core.trac.wordpress.org/browser/tags/4.9.3/src/wp-includes/default-constants.php#L0).

**Comment by jrfnl** [comment]
See #1314 ;-)

**Comment by GaryJones** [comment]
> Are you by chance scanning wp-config.php ? If so, wouldn't it make more sense to exclude that file from the constant check in this sniff ?

No, this was framework code that was doing:

```php

// Mode.
if ( ! defined( 'SCRIPT_DEBUG' ) ) {
	define( 'SCRIPT_DEBUG', false ); // @phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedConstantFound -- Valid use case as we need it defined.
}

// Assets.
define( 'BEANS_MIN_CSS', SCRIPT_DEBUG ? '' : '.min' );
define( 'BEANS_MIN_JS', SCRIPT_DEBUG ? '' : '.min' );
```

**Comment by jrfnl** [comment]
Fair enough. Anyway, the open PR should take care of it.
